go-controller/pkg/ovn/base_network_controller_pods.go

```diff
@@ -212,8 +214,17 @@ func (bnc *BaseNetworkController) deletePodLogicalPort(pod *kapi.Pod, portInfo *
 	}
 
 	shouldRelease := true
-	// check to make sure no other pods are using this IP before we try to release it if this is a completed pod.
-	if util.PodCompleted(pod) {
+	isMigratedSourcePodStale, err := kubevirt.IsMigratedSourcePodStale(bnc.watchFactory, pod)
+	if err != nil {
+		return nil, fmt.Errorf("unable to determine if ip should be released at kubevirt scenario: %v", err)
+	}
+
+	// There is a VM still running we should not deallocate the IP
+	if isMigratedSourcePodStale {
+		shouldRelease = false
+
+		// check to make sure no other pods are using this IP before we try to release it if this is a completed pod.
+	} else if util.PodCompleted(pod) {
```

Comment:
So here we prevent to release the IPs when the original pod dies but it has a migrated pod.

But when the migrated pod dies, I don't think we are releasing the IPs correctly.

For that, we should check use in this method the original switch name in case of a migrated pod and we are not doing that.

Reply:
I will call also `deleteLogicalPort` at the remote zone in case we are at kubevirt scenario, I think the logical that is not related to deallocation there do not break if stuff do not exist.